sycl/include/sycl/accessor.hpp

```diff
@@ -1217,6 +1217,41 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
 
 #endif // __SYCL_DEVICE_ONLY__
 
+public:
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_,
+            typename = detail::enable_if_t<
+                (AccessMode == access_mode::read) &&
```

Comment:
Nit; you could use `IsAccessReadOnly` instead of rewriting the check. Likewise there is a `IsAccessReadWrite`, although you may have to add a propagating definition in this class around the place where `IsAccessReadOnly` is propagated (L1018).